modules/tensor_mechanics/src/materials/ComputeReturnMappingStress.C

```diff
@@ -39,7 +39,7 @@ ComputeReturnMappingStress::initialSetup()
   std::vector<MaterialName> models = getParam<std::vector<MaterialName> >("return_mapping_models");
   for (unsigned i = 0; i < models.size(); ++i)
   {
-    MooseSharedPointer<RecomputeRadialReturn> rrr = MooseSharedNamespace::dynamic_pointer_cast<RecomputeRadialReturn>(getMaterialSharedPointerByName(models[i]));
+    MooseSharedPointer<StressUpdateBase> rrr = MooseSharedNamespace::dynamic_pointer_cast<StressUpdateBase>(getMaterialSharedPointerByName(models[i]));
```

Comment:
@permcody or @dschwen would you help me understand better how I should implement this change to raw pointers?  Is something like
`  std::string * rrr = {actually I have no idea where to look for what would replace the dynamic case}<StressUpdateBase>(getMaterialByName(models[i]))` sort of on the right track, or does the dynamic pointer cast remain with a raw pointer?